src/class_sqlThread.py

```diff
@@ -26,117 +26,134 @@
+            with open(os.path.join(root_path, "src/sql/{}.sql".format(file_name))) as sql_file:
+                sql_as_string = sql_file.read()
+            self.cur.executescript(sql_as_string)
+        except IOError as err:
+            sys.stderr.write(
+                'ERROR trying to upgrade database. Error message: %s\n' % str(err))
+            os._exit(0)
         except Exception as err:
-            if str(err) == 'table inbox already exists':
-                logger.debug('Database file already exists.')
-
-            else:
-                sys.stderr.write(
-                    'ERROR trying to create database file (message.dat). Error message: %s\n' % str(err))
-                os._exit(0)
-
-        # If the settings version is equal to 2 or 3 then the
-        # sqlThread will modify the pubkeys table and change
-        # the settings version to 4.
-        settingsversion = BMConfigParser().getint(
-            'bitmessagesettings', 'settingsversion')
-
-        # People running earlier versions of PyBitmessage do not have the
-        # usedpersonally field in their pubkeys table. Let's add it.
-        if settingsversion == 2:
-            item = '''ALTER TABLE pubkeys ADD usedpersonally text DEFAULT 'no' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 3
-
-        # People running earlier versions of PyBitmessage do not have the
-        # encodingtype field in their inbox and sent tables or the read field
-        # in the inbox table. Let's add them.
-        if settingsversion == 3:
-            item = '''ALTER TABLE inbox ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE inbox ADD read bool DEFAULT '1' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-
-            item = '''ALTER TABLE sent ADD encodingtype int DEFAULT '2' '''
-            parameters = ''
-            self.cur.execute(item, parameters)
-            self.conn.commit()
-
-            settingsversion = 4
-
-        BMConfigParser().set(
-            'bitmessagesettings', 'settingsversion', str(settingsversion))
-        BMConfigParser().save()
-
-        helper_startup.updateConfig()
+            sys.stderr.write(
+                'ERROR trying to upgrade database. Error message: %s\n' % str(err))
+            os._exit(0)
+
+    @property
+    def sql_schema_version(self):
+        return self.__get_current_settings_version()
+
+    @sql_schema_version.setter
+    def sql_schema_version(self, current_level):
+        """
+            Update version with one level
+        """
+        item = '''update settings set value=? WHERE key='version';'''
+        parameters = (current_level + 1,)
+        self.cur.execute(item, parameters)
+        self._current_level = self.__get_current_settings_version()
```

Comment:
this shouldn't be needed

Reply:
Now its 
```    @sql_schema_version.setter
    def sql_schema_version(self):
        """
            Update version with one level
        """
        item = '''update settings set value=value+1 WHERE key='version';'''
        self.cur.execute(item)
        self._current_level = self.__get_current_settings_version()
```

Reply:
Its changed into 
```        item = '''update settings set value=? WHERE key='version';'''
        self.cur.execute(item, self.__get_current_settings_version() + 1)
```
Because its not taking ```value+1```